File: admin-backend/admin-system/src/main/java/io/github/faustofan/admin/system/domain/model/SysMenu.java
package io.github.faustofan.admin.system.domain.model;

import io.github.faustofan.admin.shared.persistence.model.BaseEntity;
import io.github.faustofan.admin.system.domain.enums.MenuType;
import org.babyfish.jimmer.sql.*;
import org.jspecify.annotations.Nullable;

import java.util.List;

/**
 * 系统菜单实体
 * 表示系统中的菜单项，可以是目录、菜单或按钮
 */
@Entity
@Table(name = "sys_menu")
public interface SysMenu extends BaseEntity {

    @Key // 唯一性约束辅助
    @Nullable
    String permCode();

    String menuName();

    @Column(name = "menu_type")
    MenuType menuType(); // Enum: DIRECTORY, MENU, BUTTON

    @Nullable
    String routePath();

    @Nullable
    String componentPath();

    int sortOrder();

    boolean visible();

    @Nullable
    String icon();

    // 父子关系（递归）
    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    SysMenu parent();

    @OneToMany(mappedBy = "parent")
    List<SysMenu> children();

    // 关联：属于哪些产品包
    @ManyToMany(mappedBy = "menus")
    List<SysProductPackage> packages();

    // 关联：哪些角色拥有此菜单
    @ManyToMany(mappedBy = "menus")
    List<SysRole> roles();
}

File: src/main/kotlin/io/github/faustofan/admin/system/domain/entity/SysMenu.kt
package io.github.faustofan.admin.system.domain.entity

import io.github.faustofan.admin.common.TenantAware
import org.babyfish.jimmer.sql.*

@Entity
interface SysMenu : TenantAware {

    @Key // 同一层级下 title 不能重复，结合 parentId
    val title: String

    // 树形结构
    @ManyToOne
    @Key
    val parent: SysMenu?

    @OneToMany(mappedBy = "parent")
    val children: List<SysMenu>

    // 类型
    val type: MenuType

    // 路由路径 (如: /system/user)
    val path: String?

    // 组件路径 (如: layout/System/User/index)
    val component: String?

    // 权限标识 (如: user:add, user:delete)
    val permission: String?

    // 图标
    val icon: String?

    // 排序号
    val sortOrder: Int

    // 反向关联：拥有此菜单的角色
    @ManyToMany(mappedBy = "menus")
    val roles: List<SysRole>
}

@EnumType
enum class MenuType {
    DIRECTORY, // 目录
    MENU,      // 菜单
    BUTTON     // 按钮/权限
}